servicetalk-data-jackson-jersey3-jakarta10/build.gradle

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2018-2019, 2021-2022, 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-library"
+
+checkstyleMain {
+  enabled false
+}
+
+checkstyleTest {
+  enabled false
+}
+
+spotbugsMain {
+  enabled false
+}
+
+task cleanSources(type: Delete) {
+  delete 'src'
+}
+
+task copySourcesForJersey3(type: Copy) {
+  dependsOn tasks.cleanSources
+  from '../servicetalk-data-jackson-jersey/src'
+  into 'src'
```

Comment:
It seems like these should go into the `build/generated` folder somewhere and we can avoid having all the noise with `.gitignore` and it's more explicit what they are.
That said, I'm strugging to figure out how to do that myself. Right now I have something like this but am not able to get the tests to depend on the fixtures. The changes below are by no means complete but illustrate what I was thinking.
```
diff --git a/servicetalk-http-router-jersey3-jakarta10/build.gradle b/servicetalk-http-router-jersey3-jakarta10/build.gradle
index 72aa3bf11..674a9c791 100644
--- a/servicetalk-http-router-jersey3-jakarta10/build.gradle
+++ b/servicetalk-http-router-jersey3-jakarta10/build.gradle
@@ -24,24 +24,14 @@ checkstyleTest {
   enabled false
 }
 
-checkstyleTestFixtures {
-  enabled false
-}
-
 spotbugsMain {
   enabled false
 }
 
-task cleanSources(type: Delete) {
-  // Don't delete src/testFixtures/java/.gitkeep
-  // gitkeep used to ensure that a src/textFixtures/java exists or Gradle processing will fail
-  delete 'src/main', 'src/test', 'src/testFixtures/java/io'
-}
-
 task copySourcesForJersey3(type: Copy) {
-  dependsOn tasks.cleanSources
+//  dependsOn tasks.cleanSources
   from '../servicetalk-http-router-jersey/src'
-  into 'src'
+  into "${getBuildDir()}/generated/sources/"
   filter { line -> line.replaceAll('javax.ws.rs', 'jakarta.ws.rs') }
   filter { line -> line.replaceAll('javax.inject', 'jakarta.inject') }
 
@@ -49,11 +39,31 @@ task copySourcesForJersey3(type: Copy) {
   filter { line -> line.replaceAll('javax.annotation.Priority', 'jakarta.annotation.Priority') }
 }
 
+
+
 tasks.processResources.dependsOn(copySourcesForJersey3)
 tasks.sourcesJar.dependsOn(copySourcesForJersey3)
 tasks.compileJava.dependsOn(copySourcesForJersey3)
-tasks.processTestResources.dependsOn(copySourcesForJersey3)
-tasks.compileTestJava.dependsOn(copySourcesForJersey3)
+
+sourceSets {
+  main {
+    java {
+      srcDir "${getBuildDir()}/generated/sources/main"
+    }
+  }
+
+  testFixtures {
+    java {
+      srcDir "${getBuildDir()}/generated/sources/testFixtures"
+    }
+  }
+
+  test {
+    java {
+      srcDir "${getBuildDir()}/generated/sources/test"
+    }
+  }
+}
 
 def actualJerseyVersion = "${jersey3VersionEE10}"
 def actualJavaxActivationVersion = "${javaxActivationVersionEE10}"
diff --git a/servicetalk-http-router-jersey3-jakarta10/src/testFixtures/java/.gitkeep b/servicetalk-http-router-jersey3-jakarta10/src/testFixtures/java/.gitkeep
deleted file mode 100644
index e69de29bb..000000000
```

Reply:
It seems like it would fall into the 'build behavior' category but I don't have a problem punting that to a followup. 👍